### PR TITLE
WIP [KOGITO-8031] Kogito Serverless Platform CRD

### PR DESCRIPTION
**Pull request by davidesalerno**
Signed-off-by: Davide Salerno <dsalerno@redhat.com>

**Description of the change:**

**Motivation for the change:**

**Checklist**

- [x] Add or Modify a unit test for your change
- [x] Have you verified that tall the tests are passing?

### CODE REVIEW

File: api/v08/kogitoserverlessplatform_types.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	// the secret where credentials are stored
+	Secret string `json:"secret,omitempty"`
+	// the configmap which stores the Certificate Authority
+	CA string `json:"ca,omitempty"`
+	// the registry organization
+	Organization string `json:"organization,omitempty"`
+}
+
+type BuildSpec struct {
+	// the strategy to adopt for building a WOrkflow image
```
Comment (spolti) [inline api/v08/kogitoserverlessplatform_types.go:81]:
```suggestion
	// the strategy to adopt for building a Workflow image
```

File: api/v08/kogitoserverlessplatform_types.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	BuildPublishStrategyKaniko BuildPublishStrategy = "Kaniko"
+)
+
+// BuildPublishStrategies the list of all available publish strategies
+var BuildPublishStrategies = []BuildPublishStrategy{
+	BuildPublishStrategyKaniko,
+}
+
+// RegistrySpec provides the configuration for the container registry
+type RegistrySpec struct {
```
Comment (spolti) [inline api/v08/kogitoserverlessplatform_types.go:67]: I think we need to add the kubebuilder annotations on all objects that should be listed in the crd.

File: api/v08/kogitoserverlessplatform_types.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+	// if the container registry is insecure (ie, http only)
+	Insecure bool `json:"insecure,omitempty"`
```
Comment (spolti) [inline api/v08/kogitoserverlessplatform_types.go:69]: comments usually begins with the name of the property, e.g,. 
```
// Insecure if the container registry is insecure (ie, http only)
```

File: api/v08/kogitoserverlessplatform_types.go (hunk 1 above)
Comment (spolti) [inline api/v08/kogitoserverlessplatform_types.go:81]:
```suggestion
	// the strategy for building a Workflow container image
```

File: api/v08/kogitoserverlessplatform_types.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+	// the configmap which stores the Certificate Authority
+	CA string `json:"ca,omitempty"`
+	// the registry organization
+	Organization string `json:"organization,omitempty"`
+}
+
+type BuildSpec struct {
+	// the strategy to adopt for building a WOrkflow image
+	BuildStrategy BuildStrategy `json:"buildStrategy,omitempty"`
+	// the strategy to adopt for publishing a Workflow image
```
Comment (spolti) [inline api/v08/kogitoserverlessplatform_types.go:83]:
```suggestion
	// the strategy for publishing a Workflow image
```

File: controllers/kogitoserverlessplatform_controller.go [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	swkogitov08 "github.com/davidesalerno/kogito-serverless-operator/api/v08"
+)
+
+// KogitoServerlessPlatformReconciler reconciles a KogitoServerlessPlatform object
+type KogitoServerlessPlatformReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=sw.kogito.kie.org,resources=kogitoserverlessplatforms,verbs=get;list;watch;create;update;patch;delete
```
Comment (spolti) [inline controllers/kogitoserverlessplatform_controller.go:36]: If I recall correctly these annotations are not use in the controller

Comment (davidesalerno) [reply controllers/kogitoserverlessplatform_controller.go:48]: I think that they are used to generate the rbac configuration you can find here -> https://github.com/davidesalerno/kogito-serverless-operator/tree/main/config/rbac

**Pull request closed without merging**